common/tools/dev-tool/src/commands/run/extract-api.ts

```diff
@@ -89,62 +119,35 @@ export default leafCommand(commandInfo, async () => {
         apiReport: newApiReport,
         mainEntryPointFilePath: exports[exprt].types,
       };
-      const extractorConfig = ExtractorConfig.prepare({
-        configObject: updatedConfigObject,
-        configObjectFullPath: apiExtractorJsonPath,
-        packageJsonFullPath: packageJsonPath,
-      });
-      const extractorResult: ExtractorResult = Extractor.invoke(extractorConfig, {
-        // Equivalent to the "--local" command-line parameter
-        localBuild: true,
-        // Equivalent to the "--verbose" command-line parameter
-        showVerboseMessages: true,
-      });
-      if (extractorResult.succeeded) {
-        log.debug(`API Extractor completed successfully for ${exprt} export`);
-      } else {
-        succeed = false;
-        log.error(
-          `API Extractor completed with ${extractorResult.errorCount} errors` +
-            ` and ${extractorResult.warningCount} warnings`
-        );
-      }
+
+      succeed = extractApi(updatedConfigObject, apiExtractorJsonPath, packageJsonPath);
     }
   } else {
     // normal extraction
-    const config = ExtractorConfig.prepare({
-      configObject: extractorConfigObject,
-      configObjectFullPath: apiExtractorJsonPath,
-      packageJsonFullPath: packageJsonPath,
-    });
-
-    const extractorResult: ExtractorResult = Extractor.invoke(config, {
-      // Equivalent to the "--local" command-line parameter
-      localBuild: true,
-      // Equivalent to the "--verbose" command-line parameter
-      showVerboseMessages: true,
-    });
-    if (extractorResult.succeeded) {
-      log.debug(`API Extractor completed successfully`);
-    } else {
-      succeed = false;
-      log.error(
-        `API Extractor completed with ${extractorResult.errorCount} errors` +
-        ` and ${extractorResult.warningCount} warnings`
-      );
-    }
+    succeed = extractApi(extractorConfigObject, apiExtractorJsonPath, packageJsonPath);
   }
 
   // Add api.json files to zip archive
   const unscopedPackageName = projectInfo.name.split("/")[1];
-  const reportTempDir  = path.join(projectInfo.path, "temp");
-  const files = (await readdir(reportTempDir)).filter(f => f.endsWith("api.json"));
-  const zip = new AdmZip();
+  const reportTempDir = path.join(projectInfo.path, "temp");
+  const files = (await readdir(reportTempDir)).filter((f) => f.endsWith("api.json"));
+  const output = createWriteStream(path.join(reportTempDir, `${unscopedPackageName}.zip`));
+  const zip = archiver("zip");
+
+  zip.on("warning", function (err) {
+    throw err;
+  });
+  zip.on("error", function (err) {
+    throw err;
+  });
+
+  zip.pipe(output);
+
   for (const file of files) {
     log.debug(`adding ${file} to zip archive`);
-    zip.addLocalFile(path.join(reportTempDir, file));
+    zip.append(createReadStream(path.join(reportTempDir, file)), { name: file });
   }
-  zip.writeZip(path.join(reportTempDir, `${unscopedPackageName}.zip`));
+  zip.finalize();
```

Comment:
Should we delete the folder after the zip folder is created or failed to be created?

Reply:
The `./temp` folder exists and we just add the api.json files inside it into a zip that is also inside it.  I'd keep it this way for now to avoid breaking automations that might expect api.json file from this folder.